sound/soc/sof/topology.c

```diff
@@ -2894,6 +2920,12 @@ static int sof_route_load(struct snd_soc_component *scomp, int index,
 		goto err;
 	}
 
+	if (source_swidget->id == snd_soc_dapm_buffer) {
+		struct sof_ipc_buffer *buffer = source_swidget->private;
+		if (!buffer->size)
+			goto err;
+	}
+
```

Comment:
comment here too.

Reply:
What kind of route does this imply? a virtual VM buffer to?

Reply:
@ranj063 yes, these are virtual buffers of size 0, serving as a handle to the mixer / demux, where the actual pipeline from the guest will be attached. I'll add comments, thanks